pkg/http/client_test.go

```diff
@@ -0,0 +1,74 @@
+package http
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/dcos/dcos-cli/pkg/config"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGet(t *testing.T) {
+	query := `{"cluster":"DC/OS"}`
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte(query))
+	}))
+	defer ts.Close()
+
+	conf, err := config.FromString(`
+[core]
+
+dcos_url = "` + ts.URL + `"
+`)
+	require.NoError(t, err)
+
+	client := New(conf)
+
+	resp, err := client.Get("/")
+	require.NoError(t, err)
+
+	body, err := ioutil.ReadAll(resp.Body)
+	require.Equal(t, query, string(body))
+}
+
+func TestPost(t *testing.T) {
+	type query_struct struct {
+		Cluster string
+	}
+	query := `{"cluster":"DC/OS"}`
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "POST" {
+			t.Errorf("Expected ‘POST’ request, got ‘%s’", r.Method)
+		}
+		if r.URL.EscapedPath() != "/post" {
+			t.Errorf("expected request to ‘/post, got ‘%s’", r.URL.EscapedPath())
+		}
+
+		decoder := json.NewDecoder(r.Body)
+		var query_sent query_struct
+		err := decoder.Decode(&query_sent)
+		require.NoError(t, err)
+		defer r.Body.Close()
+
+		require.Equal(t, query_sent.Cluster, "DC/OS")
+	}))
+	defer ts.Close()
+
+	conf, err := config.FromString(`
+[core]
+
+dcos_url = "` + ts.URL + `"
+`)
+	require.NoError(t, err)
```

Comment:
Same as my previous comment, this shouldn't be the place to test config parsing.